truss/templates/shared/logging.py

```diff
@@ -1,63 +1,110 @@
 
 LEVEL: int = logging.INFO
 
-JSON_LOG_HANDLER = logging.StreamHandler(stream=sys.stdout)
-JSON_LOG_HANDLER.set_name("json_logger_handler")
-JSON_LOG_HANDLER.setLevel(LEVEL)
-JSON_LOG_HANDLER.setFormatter(
-    jsonlogger.JsonFormatter("%(asctime)s %(levelname)s %(message)s")
-)
+
+class Lifecycle(Enum):
+    LOAD = "MODEL_LOAD"
+    STARTUP = "MODEL_STARTUP"
+    REQUEST = "REQUEST"
+
+
+def serialize(record):
+    dt = datetime.fromtimestamp(record["time"].timestamp())
+    formatted_time = dt.strftime("%Y-%m-%d %H:%M:%S,%f")[:-3]
+
+    request_id = (
+        str(record["extra"]["request_id"])
+        if "request_id" in record["extra"] and record["extra"]["request_id"] is not None
+        else None
+    )
+    lifecycle = (
+        record["extra"]["lifecycle"].value
+        if "lifecycle" in record["extra"] and record["extra"]["lifecycle"] is not None
+        else None
+    )
+
+    # if someone is using logger instead of print, we need to handle that
+    try:
+        json_message = json.loads(record["message"])
+
+        # Check if 'asctime' key exists
+        if "asctime" in json_message:
+            record["message"] = json_message["message"]
+
+    except json.JSONDecodeError:
+        pass
+
+    subset = {
+        "asctime": formatted_time,
+        "message": record["message"],
+        "levelname": record["level"].name,
+        "request_id": request_id,
+        "lifecycle": lifecycle,
+    }
+
+    return json.dumps(subset)
 
 
-class HealthCheckFilter(logging.Filter):
-    def filter(self, record: logging.LogRecord) -> bool:
-        # for any health check endpoints, lets skip logging
-        return (
-            record.getMessage().find("GET / ") == -1
-            and record.getMessage().find("GET /v1/models/model ") == -1
-        )
+def patching(record):
+    record["extra"]["serialized"] = serialize(record)
 
 
-class StreamToLogger:
+loguru_logger = loguru.logger
+
+# reset the logger
+loguru_logger.remove()
+
+# we want to add request ids via a patch (think of it a middleware)
+loguru_logger = loguru_logger.patch(patching)
+loguru_logger.add(sys.stdout, format="{extra[serialized]}")
+
+
+class StreamToLogger(object):
     """
-    StreamToLogger redirects stdout and stderr to logger
+    Fake file-like stream object that redirects writes to a logger instance.
     """
 
-    def __init__(self, logger, log_level, stream):
-        self.logger = logger
-        self.log_level = log_level
+    def __init__(self, stream):
+        self.logger = loguru_logger
+        self.log_buffer = ""
         self.stream = stream
 
-    def __getattr__(self, name):
-        # we need to pass `isatty` from the stream for uvicorn
-        # this is a more general, less hacky fix
-        return getattr(self.stream, name)
-
     def write(self, buf):
-        self.logger.log(self.log_level, buf)
+        to_log = buf.rstrip()
+        if len(to_log) > 0:
+            self.logger.info(to_log)
+
+    def isatty(self):
```

Comment:
why did we need to add this?

Reply:
Added comment with info. Can't redirect stdout without it.